src/us/kbase/common/service/JsonClientCaller.java

```diff
@@ -357,108 +357,127 @@ public static AuthToken requestTokenFromKBase(
 			conn.setFixedLengthStreamingMode(size);
 		}
 		// Write real data into http output stream
-		writeRequestDataStatic(method, arg, conn.getOutputStream(), id, context);
+
+		try {
+			writeRequestDataStatic(method, arg, conn.getOutputStream(), id, context);
```

Comment:
I'd probably do
```
try (final OutputStream out = conn.getOutputStream()) {
    writeRequestDataStatic(method, arg, out, id, context);
}
```
A little more compact and lets the language handle the close.